design-docs/improved-console.md

```diff
@@ -0,0 +1,422 @@
+ * `StatusBarLine` Prefix
+ * `StatusBarLine` Suffix
+
+### Implementation
+Introduce Gradle properties that allow this customization (with defaults):
+```bash
+org.gradle.console.statusline.prefix=" > "
+org.gradle.console.statusline.suffix=""
+```
+
+### Test Cases
+* Fuzz testing ensures no other parts of build are affected by user input
+
+### Open issues
+* How is intra-operation progress ([55 / 1234] or spinner) affected?
+
+## Story: (Optional) Indicate Gradle's continued progress in absence of intra-operation updates
+Provide motion in output to indicate Gradle's working if build operations are long-running.
+
+Options:
+* Incrementing time
+* Blinking a small area of progress bar
+* Blinking indicator on progress operations
+* Adjusting color of progress operations (may require 256 colors)
+* Text-based spinner (e.g. "┘└┌┐" or "|/-\\")
+
+### Implementation
+`Operations` data store knows the time since each operation was started. The 
+`StatusBarFormatter` can choose to render an operation slightly differently depending
+on its lifespan.
+
+### Open issues
+* Best choice to render from given options above
+
+## Story: (Optional) Detect terminals that support UTF-8 and provide richer UI elements by default
+Support detection and use of UCS-2 (default supported by Windows) or UTF-8 characters could further polish this if we choose.
+
+This would allow us to use characters like ᐅ or ► supported by a majority of monospace fonts.
+More granular width for progress bars: ' ', '▏', '▎', '▎', '▍', '▍', '▌', '▌', '▋', '▋', '▊', '▊', '▉', '▉', '█'
+
+### Implementation
+Change default `org.gradle.console.progressbar.*` property values if we detect a console we know
+supports advanced features. 
+
+## Story: (Optional) "Fade out" operation result
+Tasks completed continue to be displayed for 1 more "tick" of the clock with their resolution status, 
+but are displayed in a muted color (ANSI bright black, perhaps).
+
+For example:
+```
+:api:jar UP-TO-DATE
+```
+
+## Story: (Optional) Use 256 color output when supported
+ANSI colors are limited and dependent on color mapping of the underlying terminal. 
+We can achieve much better look and feel using a 256-color palette when supported.
+
+We may also need to capture the default background color of the terminal so we 
+can adjust colors to at least light and dark backgrounds. This can be done with 
+control sequences on at least *nix terminal.
+ 
+### Implementation
+Requires use of terminfo controls. Capture output of `tput colors` when available. 
+If >=256, use a Gradle-chosen color palette.
+
+Change default `org.gradle.console.progressbar.*` property values if we detect a console we know
+supports additional colors.
+
+### Open issues
+* Color choices we would use
+
+## Story: (Optional) Illustrate build phases independently
+Similar to "Display build progress as a progress bar through colorized output" above, but having clearly separated output for each build phase.
```

Comment:
I would prioritize this over any customization. It improves several things:

- Configuration time is usually a very small part of the build, so it would only be a single tick on the progress bar. Seeing a whole bar fill up quickly instead is much more engaging.
- We don't even know how many tasks we'll execute before we calculate the task graph. By starting over we can size the increments correctly